jni/include/faiss_index_create.h

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+/**
+ * This file contains util methods which are free of JNI to be used in faiss_wrapper.cpp
+ */
+
+#ifndef OPENSEARCH_KNN_FAISS_INDEX_CREATE_H
+#define OPENSEARCH_KNN_FAISS_INDEX_CREATE_H
+
+#include "jni_util.h"
+#include "faiss/impl/io.h"
+#include "faiss/index_factory.h"
+#include "faiss/index_io.h"
+#include "faiss/IndexHNSW.h"
+#include "faiss/IndexIVFFlat.h"
+#include "faiss/MetaIndexes.h"
+#include "faiss/Index.h"
+#include "faiss/impl/IDSelector.h"
+#include "faiss/IndexIVFPQ.h"
+#include "faiss/IndexBinaryIVF.h"
+#include "faiss/IndexBinaryHNSW.h"
+#include "faiss/IndexBinary.h"
+#include <algorithm>
+#include <string>
+#include <vector>
+#include <memory>
+#include <jni.h>
+
+namespace knn_jni {
+namespace faiss_wrapper {
+
+class IndexCreateService {
```

Comment:
My only question on this would be should we make this as IndexService rather than index create service so it can support both indexing and search. We can start with create functions for now. @heemin32 whats your thoughts?

Reply:
Yes. By looking at the code again, it is badly written. Instead of having all required values as class attributes, I can simply receive them. as create method parameter.

I think factory class is not needed and the code in factory class can be in JNI method. Let me refactor it.

Reply:
Sure.. Lets give this a try. Thanks for taking up this effort. This is really tricky and was pending for a long time. Appreciate all the effort we can put to reduce the complexity